src/nccl_ofi_freelist.cpp

```diff
@@ -176,6 +189,13 @@ int nccl_ofi_freelist_fini(nccl_ofi_freelist_t *freelist)
 		size_t size = block->memory_size;
 		freelist->blocks = block->next;
 
+		for (size_t i = 0; i < block->num_entries; ++i) {
+			nccl_ofi_freelist_elem_t *entry = &block->entries[i];
+			if (freelist->entry_fini_fn) {
```

Comment:
Seems like the loop could be inside the NULL check?  Also, please explicitly check against NULL.